common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1572,3 +1572,11 @@ def set_user_partitions_for_scheme(self, partitions, scheme):
             if p.scheme != scheme
         ]
         self.user_partitions = other_partitions + partitions  # pylint: disable=attribute-defined-outside-init
+
+    def is_discussion_enabled(self, user):
+        """
+        Return True if Discussion is enabled for a course; else False
+        """
+        if not any([tab.type == 'discussion' and tab.is_enabled(self, user) for tab in self.tabs]):
```

Comment:
@wajeeha-khalid The course_overviews table can only cache the list of tab names (or the existence of the discussion tab).  It cannot, however, cache the value of `tab.is_enabled` since that checks the system-wide django settings.  So you'll need to call DiscussionTab.is_enabled in the LMS runtime, but cache the list of tabs.